resqpy/olio/wellspec_keywords.py

```diff
@@ -31,96 +32,98 @@
-wellspec_dict['RADWP'] = (0, wk_preferably_not, wk_okay, None, True)  # use pagoda wellbore radius
-wellspec_dict['PORTYPE'] = (0, wk_banned, wk_banned, None, False)  # dual porosity: todo: need to check values
-wellspec_dict['FM'] = (0, wk_preferably_not, wk_okay, 0.0, False)  # dual porosity: connection to fracture?
-wellspec_dict['MD'] = (0, wk_preferably_not, wk_okay, 'NA', False)
-wellspec_dict['PARENT'] = (0, wk_preferably_not, wk_okay, 'NA', False)
-wellspec_dict['MDCON'] = (0, wk_preferably_not, wk_okay, 'NA', False)
-wellspec_dict['SECT'] = (0, wk_preferably_not, wk_okay, 1, False)
-wellspec_dict['FLOWSECT'] = (0, wk_preferably_not, wk_okay, 1, False)
-wellspec_dict['ZONE'] = (0, wk_preferably_not, wk_okay, 1, False)
-wellspec_dict['GROUP'] = (0, wk_preferably_not, wk_okay, 1, False)
-wellspec_dict['TEMP'] = (0, wk_preferably_not, wk_okay, 'NA', False)
-wellspec_dict['IPTN'] = (0, wk_preferably_not, wk_okay, 1, False)  # pattern
-wellspec_dict['D'] = (0, wk_preferably_not, wk_okay, 'NA', False)  # non D'Arcy flow
-wellspec_dict['ND'] = (0, wk_preferably_not, wk_okay, 'NA', False)  # non D'Arcy flow
-wellspec_dict['DZ'] = (0, wk_preferably_not, wk_okay, None, True)  # non D'Arcy flow; use LENGTH value? or DZ
-
-wellspec_dtype = {}  # mapping wellspec column key to expected data type
-
-wellspec_dtype['IW'] = int
-wellspec_dtype['JW'] = int
-wellspec_dtype['L'] = int
-wellspec_dtype['GRID'] = str
-wellspec_dtype['RADW'] = float
-wellspec_dtype['KHMULT'] = float
-wellspec_dtype['STAT'] = str
-wellspec_dtype['ANGLA'] = float
-wellspec_dtype['ANGLV'] = float
-wellspec_dtype['LENGTH'] = float
-wellspec_dtype['KH'] = float
-wellspec_dtype['SKIN'] = float
-wellspec_dtype['PPERF'] = float
-wellspec_dtype['ANGLE'] = float
-wellspec_dtype['IRELPM'] = int
-wellspec_dtype['RADB'] = float
-wellspec_dtype['WI'] = float
-wellspec_dtype['K'] = float
-wellspec_dtype['LAYER'] = int
-wellspec_dtype['DEPTH'] = float  # '#' causes nexus to use cell depth
-wellspec_dtype['X'] = float  # use cell X for i.p. perf
-wellspec_dtype['Y'] = float  # use cell Y for i.p. perf
-wellspec_dtype['CELL'] = int  # CELL is for unstructured grids
-wellspec_dtype['DTOP'] = float  # not compatible with ANGLA, ANGLV
-wellspec_dtype['DBOT'] = float  # not compatible with ANGLA, ANGLV
-wellspec_dtype['RADBP'] = float  # calculate as for RADB
-wellspec_dtype['RADWP'] = float
-wellspec_dtype['PORTYPE'] = str  # dual porosity: todo: need to check type
-wellspec_dtype['FM'] = float
-wellspec_dtype['MD'] = float
-wellspec_dtype['PARENT'] = str
-wellspec_dtype['MDCON'] = float
-wellspec_dtype['SECT'] = str  # todo: need to check type
-wellspec_dtype['FLOWSECT'] = str  # todo: need to check type
-wellspec_dtype['ZONE'] = int
-wellspec_dtype['GROUP'] = str
-wellspec_dtype['TEMP'] = float
-wellspec_dtype['IPTN'] = int
-wellspec_dtype['D'] = float
-wellspec_dtype['ND'] = str
-wellspec_dtype['DZ'] = float
+# yapf: disable
```

Comment:
FYI @andy-beer - I thought here was a good place to try out the `# yapf: disable` to keep the previous indentation, seems to work as expected.

Reply:
Yes, that looks good.

I think most of the data in wellspec_dict is only used by pagoda and probably should be stripped out of resqpy anyway!  However, I can look into that separately at a later date.